lib/src/value/color.ts

```diff
@@ -3,371 +3,1112 @@
-  //
-  // Algorithm from the CSS3 spec: https://www.w3.org/TR/css3-color/#hsl-color.
-  private hslToRgb(): void {
-    const scaledHue = this.hue / 360;
-    const scaledSaturation = this.saturation / 100;
-    const scaledLightness = this.lightness / 100;
+  /**
+   * Returns a new color that's the result of changing one or more of this
+   * color's channels.
+   */
+  change(
+    options: {
+      [key in ChannelNameHsl]?: number | null;
+    } & {
+      space?: ColorSpaceHsl;
+    }
+  ): SassColor;
+  change(
+    options: {
+      [key in ChannelNameHwb]?: number | null;
+    } & {
+      space?: ColorSpaceHwb;
+    }
+  ): SassColor;
+  change(
+    options: {
+      [key in ChannelNameLab]?: number | null;
+    } & {
+      space?: ColorSpaceLab;
+    }
+  ): SassColor;
+  change(
+    options: {
+      [key in ChannelNameLch]?: number | null;
+    } & {
+      space?: ColorSpaceLch;
+    }
+  ): SassColor;
+  change(
+    options: {
+      [key in ChannelNameRgb]?: number | null;
+    } & {
+      space?: ColorSpaceRgb;
+    }
+  ): SassColor;
+  change(
+    options: {
+      [key in ChannelNameXyz]?: number | null;
+    } & {
+      space?: ColorSpaceXyz;
+    }
+  ): SassColor;
+  change(
+    options: {
+      [key in ChannelName]?: number | null;
+    } & {
+      space?: KnownColorSpace;
+    }
+  ) {
+    const spaceSetExplicitly = !!options.space;
+    let space = options.space ?? this.space;
+    if (this.isLegacy && !spaceSetExplicitly) {
+      if (
+        propertyIsSet(options.whiteness) ||
+        propertyIsSet(options.blackness) ||
+        (this.space === 'hwb' && propertyIsSet(options.hue))
+      ) {
+        space = 'hwb';
+      } else if (
+        propertyIsSet(options.hue) ||
+        propertyIsSet(options.saturation) ||
+        propertyIsSet(options.lightness)
+      ) {
+        space = 'hsl';
+      } else if (
+        propertyIsSet(options.red) ||
+        propertyIsSet(options.green) ||
+        propertyIsSet(options.blue)
+      ) {
+        space = 'rgb';
+      }
+      if (space !== this.space) {
+        emitColor4ApiChangeSpaceDeprecation();
+      }
+    }
+
+    // Validate channel values
+    const keys = Object.keys(options).filter(
+      key => key !== 'space'
+    ) as ChannelName[];
+    for (const channel of keys) {
+      validateChannelInSpace(channel, space);
+    }
+    if (propertyIsSet(options.alpha) && options.alpha !== null) {
+      fuzzyAssertInRange(options.alpha, 0, 1, 'alpha');
+    }
+    if (propertyIsSet(options.lightness) && options.lightness !== null) {
+      const maxLightness = space === 'oklab' || space === 'oklch' ? 1 : 100;
+      assertClamped(options.lightness, 0, maxLightness, 'lightness');
+    }
+
+    const color = this.toSpace(space);
+    const getChangedValue = (channel: ChannelName) => {
+      if (propertyIsSet(options[channel])) {
+        return options[channel];
+      }
+      return color.channel(channel);
+    };
+    let changedColor: SassColor;
```

Comment:
Consider moving the switch below into a helper as well so you can just `return this.changeHelper(...).toSpace(...)`.

Reply:
Done in https://github.com/sass/embedded-host-node/pull/259/commits/18ab10ee84ce3b3321b8a8b98f52c7fb79650653